### ISSUE TITLE
[Bug] lax.isnan and lax.isinf do not handle NaN values correctly

### ISSUE THREAD

**Issue by neerajprad** [google/jax#545]
I think the issue can be traced down to `lax.eq(np.inf, np.nan)` returning True, for some reason. I am on a recent master commit (7328689).

```python
>>> import jax.numpy as np
>>> from jax.numpy.lax_numpy import isinf
>>> from jax import lax
>>> nan = np.array([np.nan])
>>> print(nan)
 array([nan], dtype=float32)
>>> np.isnan(nan)  # Should be True
 array([False])
>>> np.isinf(nan) # Should be False
 array([True])
>>> lax.eq(np.nan, np.inf)  # Should be False
 array(True)
```

**Comment by mattjj** [comment]
This might be a duplicate of #276, which is frustrating but our progress is blocked on upstream XLA and/or LLVM fixes.

Can you take a look at that issue and see if you think it's the same one?

**Comment by neerajprad** [comment]
> Can you take a look at that issue and see if you think it's the same one?

Thanks, @mattjj. You are absolutely right, disabling fast math mode fixed it! I will close this issue.
